docs/content/guides/dagster/pythonic-resources.mdx

```diff
@@ -0,0 +1,337 @@
+---
+title: Pythonic resources | Dagster
+description: Pythonic resources are an experimental new feature that makes Dagster Resources easier to use and more powerful.
+---
+
+# Pythonic resources
+
+<Note>
+  This feature is considered <strong>experimental</strong>.
+</Note>
+
+This guide acts as an introduction to Dagster resources utilizing the new Pythonic resources API layer, which makes defining and using Dagster [resources](/concepts/resources) easier.
+
+Resources are objects that are shared across the implementations of multiple [software-defined assets](/concepts/assets/software-defined-assets) and [ops](/concepts/ops-jobs-graphs/ops) and that can be plugged in after defining those ops and assets.
+
+Resources typically model external components that assets and ops interact with. For example, a resource might be a connection to a data warehouse like Snowflake or a service like Slack.
+
+So, why use resources?
+
+- **Plug in different implementations in different environments** - If you have a heavy external dependency that you want to use in production, but avoid using in testing, you can accomplish this by providing different resources in each environment. Check out [Separating Business Logic from Environments](/concepts/testing#separating-business-logic-from-environments) for more info about this capability.
+- **Share configuration across multiple ops or assets** - Resources are configurable and shared, so you can supply configuration in one place instead of configuring the ops and assets individually.
+- **Share implementations across multiple ops or assets** - When multiple ops access the same external services, resources provide a standard way to structure your code to share the implementations.
+
+---
+
+## Using resources
+
+### Getting started using resources
+
+A resource can be any Python object which you would like to share between software-defined assets or ops. Assets and ops specify resource dependencies by annotating the resource as a parameter to the asset or op function. The parameter type should be wrapped in a <PyObject object="Resource"/> annotation to indicate that it is a resource and not an upstream asset or op input.
```

Comment:
`The parameter type should be wrapped in a <PyObject object="Resource"/> annotation to indicate that it is a resource and not an upstream asset or op input.`

This is going to present a challenge for sure. I consider this an advanced capability but you are right that users will probably encounter it early in their journey. Still I would much prefer to try to introduce the `Resource` annotation later and instead assume they will annotate with their resource definition

Reply:
moved out of critical path to an 'advanced usage' section